packages/flutter/lib/src/widgets/routes.dart

```diff
@@ -841,11 +843,11 @@ class _ModalScopeState<T> extends State<_ModalScope<T>> {
                             key: widget.route._subtreeKey, // immutable
                             child: Builder(
                               builder: (BuildContext context) {
-                                return widget.route.buildPage(
+                                return _FocusTrap(child: widget.route.buildPage(
```

Comment:
any reason we scope at route level?

Reply:
i scoped it at the route level so that I can have this and the shortcuts widget used by the SelectionArea widget use the same focus node.